### PR TITLE
For #17917: Use View binding in Login Exceptions

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
For https://github.com/mozilla-mobile/fenix/issues/17917

### Pull Request checklist

- [x] **Tests**: This PR includes no tests as it is only a minor change
- [x] **Screenshots**: This PR does not include screenshots as no user visible changes have been made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/logins/fragment/LoginDetailFragment.kt [hunk 1]
```diff
@@ -77,7 +77,8 @@ class LoginDetailFragment : Fragment(R.layout.fragment_login_detail) {
             )
         }
         loginDetailView = LoginDetailView(
-            view.findViewById(R.id.loginDetailLayout)
+            view.findViewById(R.id.loginDetailLayout),
+            binding
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/settings/logins/fragment/LoginDetailFragment.kt:81]: This seems strange, but again this `LoginDetailView` is not really a View but rather a delegate to handle binding.

**Review by Mugurell — DISMISSED** [review]

File: app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt [hunk 2]
```diff
@@ -21,21 +20,30 @@ abstract class ExceptionsView<T : Any>(
     protected val interactor: ExceptionsInteractor<T>
 ) : LayoutContainer {
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: nit: I think we don't need `LayoutContainer` as we are using view binding :)

Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: I blame all of this on @Mugurell

Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: Right. This is a `android.view.View/ViewGroup`but not a `kotlinx.android.extensions.LayoutContainer`.

Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: We'll have to make sure that when the migration ends there is also no `kotlinx.android.extensions` import anywhere in the project.

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: Maybe I'm missing something :), but I think the import is from  `kotlinx.android.extensions.LayoutContainer`

image

Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: @Amejia481 Agree. And when the migration ends there should't be any trace of this import in the project. Sorry, I think we were overlapping in comments.
I was saying that this is a `android.view.ViewGroup` (which receives the parent layout in which it inflates and attaches itself) and can be declared as such but is not a `kotlinx.android.extensions.LayoutContainer` so this is a something that should be refactored as part of this migration work.

Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: Removing `LayoutContainer` breaks `TrackingProtectionExceptionsView`. Unsure what to do next.

Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: Same changes I here, just making the `binding` property from `ExceptionsView` protected to be available in `TrackingProtectionExceptionsView` seems to work for me.

Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsView.kt:21]: Then I'll have to wait for https://github.com/mozilla-mobile/fenix/pull/20788/ to merge

**Comment by codrut-topliceanu** [comment]
Added `do-not-land` label to wait for trackingProtection PR to land. (WIP)

**Review by Mugurell — APPROVED** [review]
LGTM.

**Pull request merged**